### PR TITLE
Add IAssemblySymbol.GetForwardedTypes API.

### LINKED ISSUE #36274: Provide API for IAssemblySymbol to retrieve type forwards
There's already an internal API on PEModuleSymbol, GetForwardedTypes, but I don't think it's ready to be made public. We should actually do some design work to figure out how we want this API to behave.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Closes #36274

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review.

File: src/Compilers/Core/Portable/Symbols/IAssemblySymbol.cs [hunk 1]
```diff
@@ -76,6 +77,11 @@ public interface IAssemblySymbol : ISymbol
         /// </summary>
         INamedTypeSymbol? ResolveForwardedType(string fullyQualifiedMetadataName);
 
+        /// <summary>
+        /// Returns type symbols for top-level (non-nested) types forwarded by this assembly.
```
[the remark this answers is not in the document]
Comment (AlekseyTs) [reply src/Compilers/Core/Portable/Symbols/IAssemblySymbol.cs:81]: > How do nested types participate in type forwarding, if they participate at all?

Compilers emit metadata information describing forwarding of nested types. However, compilers themselves don't use this information because a nested type cannot be forwarded in a different place, it should still be inside the same containing type. So, they necessarily move together and if you locate the container, no need to go through a complicated lookup for a nested type. Also, compilers don't even allow using a nested type as an argument for TypeForwardedTo attribute. 
Bottom line:
- I believe the information about forwarded nested types is redundant.
- Including that information would require more changes within compilers and as a result would increase implementation and testing cost.

---
In reply to: [426731695](https://github.com/dotnet/roslyn/pull/44307#discussion_r426731695) [](ancestors = 426731695)

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review.

File: src/Compilers/CSharp/Portable/Symbols/PublicModel/AssemblySymbol.cs [hunk 2]
```diff
@@ -55,6 +55,12 @@ INamedTypeSymbol IAssemblySymbol.ResolveForwardedType(string fullyQualifiedMetad
             return UnderlyingAssemblySymbol.ResolveForwardedType(fullyQualifiedMetadataName).GetPublicSymbol();
         }
 
+        ImmutableArray<INamedTypeSymbol> IAssemblySymbol.GetForwardedTypes()
+        {
+            return UnderlyingAssemblySymbol.GetAllTopLevelForwardedTypes().Select(t => t.GetPublicSymbol()).
+                   OrderBy(t => t.ToDisplayString(SymbolDisplayFormat.QualifiedNameArityFormat)).AsImmutable();
```
Comment (333fred) [inline src/Compilers/CSharp/Portable/Symbols/PublicModel/AssemblySymbol.cs:61]: >OrderBy [](start = 19, length = 7)

Nit: consider putting the `.` on this line. Having it on the previous line and this line starting with what appears to be receiver-less invocation is confusing.

Comment (AlekseyTs) [reply src/Compilers/CSharp/Portable/Symbols/PublicModel/AssemblySymbol.cs:61]: I am coming from school that leaves operators on the previous line.

---
In reply to: [426949277](https://github.com/dotnet/roslyn/pull/44307#discussion_r426949277) [](ancestors = 426949277)
